File: src/SocialApp/src/app/auth/models/user-dto.ts
export interface UserDTO {
  id: number;
  firstName: string;
  lastName: string;
  gender: Gender;
  age: number;
  created: string;
  lastActive: string;
  bio: string;
  relationStatus: RelationStatus;
  pictures: PictureDTO[];
}

export enum Gender {
  NotSpecified = 0,
  Male = 1,
  Female = 2
}

export enum RelationStatus {
  None = 0,
  Friend = 1,
  FriendRequested = 2,
  FriendRequestReceived = 3
}

export interface PictureDTO {
  id: number;
  url: string;
}

File: src/API/Application/Features/Users/UserDTO.cs
namespace Application.Features.Users;

public record UserDTO
{
    public int Id { get; init; }
    public required string FirstName { get; init; }
    public required string LastName { get; init; }
    public Gender Gender { get; init; }
    public int Age { get; init; }
    public DateTime Created { get; init; }
    public DateTime LastActive { get; init; }
    public required string Bio { get; init; }
    public RelationStatus RelationStatus { get; init; } = RelationStatus.None;
    public required string? ProfilePictureUrl { get; init; }
    public IEnumerable<PictureDTO> Pictures { get; init; } = [];
}
